### DISCUSSION TITLE
citation of the library in published work

### DISCUSSION THREAD

**Discussion by DavidePatria** [do-mpc/do-mpc#160] [General]
Hello,
I'm about to publish a paper that has been developed using this library and I saw what the readme mentions regarding citing the library, but just below that it says to properly mention the other used softwares.
where could I get a full list of these so that my citation could be complete?
thank you

**Comment by 4flixt** [comment]
Hi @DavidePatria,

first of all thank you for sharing this, we are happy that do-mpc helps other researchers :) 
Please cite also[ CasADi](https://web.casadi.org/publications/) and [IPOPT](https://coin-or.github.io/Ipopt/).

You could write something like:

> For the implementation we used do-mpc (cite us), a toolbox for non-linear robust MPC based on CasADI (cite) and IPOPT (cite).

Fingers are crossed for your work!

Best
4flixt

**Comment by DavidePatria** [comment]
the article has been published. thank you for providing such an amazing tool. Vielen dank

**Reply by 4flixt** [reply]
Hey congrats :) Is a pre-print available? We would be very interest to read it :)

**Reply by DavidePatria** [reply]
sure, [here it is](https://www.mdpi.com/1424-8220/21/12/4193).

edit: it is based on my [master thesis](https://webthesis.biblio.polito.it/18254/)
